### PR TITLE
[DPE-4874] [DPE-4868] [DPE-4870] Update libs for log rotate

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Problem
The code for supporting log rotate is on the shared mongodb libs for `v1` - this charm uses mostly `v0` libs

## Solution
1. Update the necessary libs for log rotation to `v1`
2. Make corresponding changes in the charm such that these changes do not break the charm
3. 1+2 break tests, update them. 

## Future work
1. Update all remaining libs to their associated `v1` version
2. Implement log rotate
3. Further improve Ha tests once log rotate is fully added

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -55,3 +55,6 @@ storage:
   mongodb:
     type: filesystem
     location: /var/lib/mongodb
+  mongodb-logs:
+    type: filesystem
+    location: /var/log/mongodb
```
Comment (MiaAltieri) [inline metadata.yaml:60]: should probably decide what we want to do here before merging

File: src/charm.py [hunk 2]
```diff
@@ -1196,6 +1201,63 @@ def restart_backup_service(self) -> None:
         container = self.unit.get_container(Config.CONTAINER_NAME)
         container.restart(Config.Backup.SERVICE_NAME)
 
+    def check_relation_broken_or_scale_down(self, event: RelationDepartedEvent) -> None:
```
Comment (MiaAltieri) [inline src/charm.py:1204]: this chunk was copied from MongoDB charm - maybe a future todo would be to make this into a lib?

**Review by Gu1nness — APPROVED** [review]

File: src/charm.py (hunk 2 above)
Comment (Gu1nness) [reply src/charm.py:1204]: Agreed, but as it uses `unit_peer_data`, either we need to enforce the naming, or bring it in the library, or find another way

**Review by Mehdi-Bendriss — APPROVED** [review]

File: src/charm.py (hunk 2 above)
Comment (MiaAltieri) [reply src/charm.py:1204]: Agreed - espcially then we can copy

**Pull request merged**